### DISCUSSION TITLE
Render as image without a file?

### DISCUSSION THREAD

**Discussion by soulflyman** [zachzurn/thermal#24] [Q&A]
Hi is there already a possibility to render printer code as an image but keep the image in the memory and don't write it to a file.
This would allow me to save the data into a database instead on the file system.
Currently, I render the code, then it gets saved into a file, I then read the file and write the content to the database.

Eliminating the temporary file would be more efficient.

**Comment by zachzurn** [comment]
The current image renderer doesn't support that but it can be added. I think adding a callback as a second argument to the image renderer could do the trick, then you would get a callback with bytes instead of writing to file.

On another note, it's pretty common to just store a url to a file in the database vs storing the image file content.

**Reply by zachzurn** [reply]
We have overhauled the rendering system to allow for this.
